src/sentry/integrations/slack/utils/channel.py

```diff
@@ -120,56 +121,89 @@ def get_channel_id_with_timeout(
         "types": "public_channel,private_channel",
     }
 
-    list_types = LIST_TYPES
-
     time_to_quit = time.time() + timeout
 
     client = SlackClient(integration_id=integration.id)
     id_data: Optional[Tuple[str, Optional[str], bool]] = None
     found_duplicate = False
+
+    token = integration.metadata.get("user_access_token") or integration.metadata["access_token"]
+    id_data: Optional[Tuple[str, Optional[str], bool]] = None
+    found_duplicate = False
     prefix = ""
-    for list_type, result_name, prefix in list_types:
-        cursor = ""
-        while True:
-            endpoint = f"/{list_type}.list"
-            try:
+
+    cursor = ""
+
+    try:  # Check for channel
+        msg_response = client.post(
```

Comment:
@chloeho7 Maybe we can put both `post` requests in some function that returns `msg_response["channel"]` to make this code a bit cleaner